extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt

```diff
@@ -56,7 +56,14 @@ internal class CarMapSurfaceOwner(
     logI(TAG, "surfaceAvailable")
     val oldCarMapSurface = this.mapboxCarMapSurface
     this.mapboxCarMapSurface = mapboxCarMapSurface
-    oldCarMapSurface?.let { carMapObservers.forEach { it.onDetached(oldCarMapSurface) } }
+    oldCarMapSurface?.let {
+      carMapObservers.forEach { it.onDetached(oldCarMapSurface) }
+      // The surface can become available without a call to surfaceDestroyed
+      // https://issuetracker.google.com/issues/235121269
```

Comment:
do we need to revert this change after the upstream issue is fixed? If yes, we should probably open a new issue to track the tailwork.

Reply:
I'm not sure if we will ever revert it, I think we should first wait for a response on the issuetracker.

There have been reports of this happening in production for android auto head units, where the surface is re-created without being destroyed. This may be a safeguard that we keep, considering it is supporting both AA and AAOS